### PR TITLE
Unit tests for `cloud/pkg/cloudhub/common/helper.go`

### LINKED ISSUE #4765: increase test coverage for `application.go`
**What would you like to be added/modified**:
I would like to add unit test for `pkg/metaserver/application.go`
**Why is this needed**:

### PR DESCRIPTION
**Pull request by 1Shubham7**
**What type of PR is this?**

/kind test

**What this PR does / why we need it**:

I have added unit tests for `cloud/pkg/cloudhub/common/helper.go`

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cloud/pkg/cloudhub/common/helper_test.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+	"github.com/kubeedge/kubeedge/common/constants"
+	"encoding/json"
+
+	beehivemodel "github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/model"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	edgecon "github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/constants"
+	// "k8s.io/apimachinery/pkg/runtime"
+	// "k8s.io/apimachinery/pkg/runtime/schema"
```
Comment (WillardHu) [inline cloud/pkg/cloudhub/common/helper_test.go:31]: Please optimize the grouping of imported packages, and commented codes can be considered for deletion.

**Comment by WillardHu** [comment]
Thanks for your contribution, please provide the test results(Inclusion coverage)

File: cloud/pkg/cloudhub/common/helper_test.go (hunk 1 above)
Comment (1Shubham7) [reply cloud/pkg/cloudhub/common/helper_test.go:31]: oh noo, really sorry. fixing it

**Comment by 1Shubham7** [comment]
> Thanks for your contribution, please provide the test results(Inclusion coverage)

This is the test coverage (for two files):

```
go tool cover -func=coverage.out
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:45:            IsVolumeResource                100.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:50:            GetMessageUID                   100.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:60:            GetMessageDeletionTimestamp     100.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:72:            TrimMessage                     100.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:84:            ConstructConnectMessage         88.9%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:102:           DeepCopy                        85.7%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:113:           NotifyEventQueueError           0.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:123:           AckMessageKeyFunc               100.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/helper.go:136:           NoAckMessageKeyFunc             100.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/message_pool.go:48:      InitNodeMessagePool             0.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/message_pool.go:58:      GetAckMessage                   0.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/message_pool.go:81:      GetNoAckMessage                 0.0%
github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common/message_pool.go:104:     ShutDown                        0.0%
total:                                                                          (statements)                    55.7%

```


Test results:

```
go test -v
=== RUN   TestIsVolumeResource
--- PASS: TestIsVolumeResource (0.00s)
=== RUN   TestGetMessageUID
=== RUN   TestGetMessageUID/Valid_UID
=== RUN   TestGetMessageUID/Invalid_content_type
--- PASS: TestGetMessageUID (0.00s)
    --- PASS: TestGetMessageUID/Valid_UID (0.00s)
    --- PASS: TestGetMessageUID/Invalid_content_type (0.00s)
=== RUN   TestGetMessageDeletionTimestamp
=== RUN   TestGetMessageDeletionTimestamp/Valid_DeletionTimestamp
=== RUN   TestGetMessageDeletionTimestamp/Invalid_content_type
--- PASS: TestGetMessageDeletionTimestamp (0.00s)
    --- PASS: TestGetMessageDeletionTimestamp/Valid_DeletionTimestamp (0.00s)
    --- PASS: TestGetMessageDeletionTimestamp/Invalid_content_type (0.00s)
=== RUN   TestTrimMessage
=== RUN   TestTrimMessage/Valid_resource
=== RUN   TestTrimMessage/Invalid_resource_length
W0620 06:41:05.066615   27032 helper.go:77] event resource node/nodeName starts with node but length less than 3
=== RUN   TestTrimMessage/Resource_not_starting_with_node
--- PASS: TestTrimMessage (0.00s)
    --- PASS: TestTrimMessage/Valid_resource (0.00s)
    --- PASS: TestTrimMessage/Invalid_resource_length (0.00s)
    --- PASS: TestTrimMessage/Resource_not_starting_with_node (0.00s)
=== RUN   TestConstructConnectMessage
--- PASS: TestConstructConnectMessage (0.00s)
=== RUN   TestDeepCopy
--- PASS: TestDeepCopy (0.00s)
=== RUN   TestAckMessageKeyFunc
=== RUN   TestAckMessageKeyFunc/Valid_message_with_GroupResource
=== RUN   TestAckMessageKeyFunc/Invalid_object_type
=== RUN   TestAckMessageKeyFunc/Message_without_GroupResource
--- PASS: TestAckMessageKeyFunc (0.00s)
    --- PASS: TestAckMessageKeyFunc/Valid_message_with_GroupResource (0.00s)
    --- PASS: TestAckMessageKeyFunc/Invalid_object_type (0.00s)
    --- PASS: TestAckMessageKeyFunc/Message_without_GroupResource (0.00s)
=== RUN   TestNoAckMessageKeyFunc
=== RUN   TestNoAckMessageKeyFunc/Valid_message
=== RUN   TestNoAckMessageKeyFunc/Invalid_object_type
--- PASS: TestNoAckMessageKeyFunc (0.00s)
    --- PASS: TestNoAckMessageKeyFunc/Valid_message (0.00s)
    --- PASS: TestNoAckMessageKeyFunc/Invalid_object_type (0.00s)
PASS
ok      github.com/kubeedge/kubeedge/cloud/pkg/cloudhub/common  0.028s
```

File: cloud/pkg/cloudhub/common/helper_test.go [hunk 2]
```diff
@@ -0,0 +1,267 @@
+/*
+Copyright 2022 The KubeEdge Authors.
```
Comment (Shelley-BaoYue) [inline cloud/pkg/cloudhub/common/helper_test.go:2]:
```suggestion
Copyright 2024 The KubeEdge Authors.
```

**Review by Shelley-BaoYue — COMMENTED** [review]
Maybe we need to wait PR #5675 (introduce "github.com/stretchr/testify/assert" )merged first

File: cloud/pkg/cloudhub/common/helper_test.go [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+				Content: "",
+			},
+			stdResult: "",
+			hasError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := GetMessageUID(tt.msg)
```
Comment (wbc6080) [inline cloud/pkg/cloudhub/common/helper_test.go:71]:
```suggestion
	for _, case := range tests {
		t.Run(case.name, func(t *testing.T) {
			result, err := GetMessageUID(case.msg)
```
It seems like the `tt` variable here refers to the test case? Maybe we can use the name `case`, which is easier to understand than `tt`. Could other similar names be changed to be easier to understand?

Comment (1Shubham7) [reply cloud/pkg/cloudhub/common/helper_test.go]: yes, `tt` refers to test case. I have now replaced it with the conventional naming which is easier to understand. I have ensured to use names that are easier to understand already but now I also double checked. Please let me know if I should make any other changes.

**Comment by Shelley-BaoYue** [comment]
As PR #5675 has been merged, you need rebase the master branch and push again.

**Review by Shelley-BaoYue — COMMENTED** [review]
I think you could squash these commits together. That would make it easier for us to review or search through commit history.

**Pull request closed without merging**